tfx/experimental/pipeline_testing/dummy_launcher_test.py

```diff
@@ -0,0 +1,175 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for tfx.components.trainer.executor."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import absl
+from ml_metadata.proto import metadata_store_pb2
+import mock
+import os
+from typing import Any, Dict, List, Text
+
+import tensorflow as tf
+from tfx import types
+from tfx.experimental.pipeline_testing import dummy_component_launcher
+from tfx.experimental.pipeline_testing import dummy_executor
+from tfx.orchestration import data_types
+from tfx.orchestration import metadata
+from tfx.orchestration import publisher
+from tfx.orchestration.launcher import test_utils
+from tfx.types import channel_utils
+from tfx.utils import io_utils
+
+class CustomDummyExecutor(dummy_executor.BaseDummyExecutor):
+  def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+         output_dict: Dict[Text, List[types.Artifact]],
+         exec_properties: Dict[Text, Any]) -> None:
+    absl.logging.info("Running CustomDummyExecutor")
+    for artifact_list in output_dict.values():
+      for artifact in artifact_list:
+        custom_output_path = os.path.join(artifact.uri, "result.txt")
+        tf.io.gfile.makedirs(os.path.dirname(custom_output_path))
```

Comment:
Do we need to make this dirs? It looks like handled in `write_string_file()` function.